### ISSUE TITLE
Connection to device succeeded but no datapoints found,

### ISSUE THREAD

**Issue by lcsneil** [rospogrigio/localtuya#1468]
Tuya Clamp Power Meter PJ-MGW1103
## The problem

Device connects fine to Tuya Cloud and works with the phone app no issues. Try to then configure it on localtuya.
The device is found and I have put in the parameters from the web account and get error that device is found but no data points. 

## Environment
 Using Version 5.2.1 (have also tried 5.0.0 with same results)
HA running on RPi4

- Localtuya version: 5.2.1 
- Home Assistant Core version:  
- [] Does the device work using the Home Assistant Tuya Cloud component ? X
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [] Was the device working with earlier versions of localtuya ? Which one? 
- [] Are you using the Tuya/SmartLife App in parallel ? X

## Provide Home Assistant traceback/logs

```
Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:450
Integration: LocalTuya integration ([documentation](https://github.com/rospogrigio/localtuya/), [issues](https://github.com/rospogrigio/localtuya/issues))
First occurred: 07:47:41 (1 occurrences)
Last logged: 07:47:41

[bf9...9hp] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 490, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/locks.py", line 387, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 492, in wait_for
    raise exceptions.TimeoutError() from exc
TimeoutError
```

## Additional information

![localTuya](https://github.com/rospogrigio/localtuya/assets/35435828/99df12c4-dbf3-402b-88bf-f70e611fec80)

**Comment by e3drigs** [comment]
Sharing this problem trying to set up localtuya for first time today.
HA 2023.7.3
localTuya 5.2.1

Attempting to add heaters and bulbs both yield same message and stack trace.

**Comment by gh69** [comment]
I have had this same issue for nearly a year now on multiple versions.  I'm currently using HA core 2023.73 and localTuya 5.3.1 and tried getting it working again (since my free year of Tuya cloud is almost up) but still seeing the same problem.  I only have 1 Tuya compatible device on my account and on my network.  It is a pet feeder, which has a toggle switch as well as a camera.  This device works perfectly with the cloud based Tuya integration but I have not been able to get it to work on Local Tuya.  i've got the device id and the looked-up local_key but when I attempt to add the device using protocol version 3.3, I *immediately* get the error "Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists."  If I attempt to add the device with protocol version 3.4, it spins there for a minute before coming back with that same error.

**Comment by gootr00** [comment]
I have the sam problem. I cannot get the camera's in homeassistant tuya nor localtuya.

**Comment by gh69** [comment]
I actually have no problems with the camera using the regular cloud-based tuya integration.  With the regular cloud-based tuya integration the cam shows up as a sensor and all of it's extras show up as configuration controls (flip [switch], motion alarm [switch], motion detection sensitivity [drop down list], motion recording [switch], record mode [drop down list], video recording [switch]) and they all work great.  The localtuya integration is where I'm seeing the no-datapoints-found error and this is preventing me from using the camera and replacing the cloud-based tuya integration with the localtuya integration.

**Comment by Justmeagene** [comment]
Same here cloud-based tuya works but got the same error as lcsneil in the local tuya

hope its fixet soon

**Comment by imadegunawinangun** [comment]
Im facing same issue

**Comment by shazn786** [comment]
i am also getting this same error while trying to add a breaker with energy monitor feature, whereas i am using 7 other on/off switches with energy monitor (the switches work fine while also giving me readings regarding current, voltage and watts usage)

[config_entry-localtuya-0f4d0723e6185c802cc31923261f5da4.json.txt](https://github.com/rospogrigio/localtuya/files/12326508/config_entry-localtuya-0f4d0723e6185c802cc31923261f5da4.json.txt)

[localtuya-0f4d0723e6185c802cc31923261f5da4-Mains Breaker-EM-1141bd2eb8eaf4a4c7747b146b792e97.json.txt](https://github.com/rospogrigio/localtuya/files/12326509/localtuya-0f4d0723e6185c802cc31923261f5da4-Mains.Breaker-EM-1141bd2eb8eaf4a4c7747b146b792e97.json.txt)

**Comment by meierthomas** [comment]
**While local-tuya works fine for me for my Wifi smart dimmers** I have the same issues reported here when trying to add an outdoor RGB LED; i.e.

`Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.`

Log file of the process of adding the RGB LED
[home-assistant_localtuya_2023-08-13T01-44-40.548Z.log](https://github.com/rospogrigio/localtuya/files/12328606/home-assistant_localtuya_2023-08-13T01-44-40.548Z.log)

I can add the device in the Tuya Smart app and it works fine. When adding it to Home Assistant I get the above error.

Did a bit more digging around datapoints. I have used the method described by Jimmy James (https://www.youtube.com/watch?v=Q1ZShFJDvE0&t=372s) to find the keys. In the referenced time stamp he also shows how to check the info on data points from the IOT cloud service. I have done the same with the RGB light and I don't get a `dp_id:` value as shown in the video from Jimmy James; i.e.

![image](https://github.com/rospogrigio/localtuya/assets/24826786/17957022-ab82-4bbc-9351-baa2624583ff)

What I get from my light is the following specs; you see that `dp_id` is missing from my RGB LED

![image](https://github.com/rospogrigio/localtuya/assets/24826786/93a877ad-fec5-4614-aad4-553cc88666e1)

So I wonder whether there's a firmware issue or a setting to make this available...

**Comment by Milliamper** [comment]
Same issue here too with PJ-1103 WiFi Digital Meter
- I have tried Local Tuya integration, with and without Cloud API
- I also tried downloading and using the Local Tuya master branch
- The Tuya integration shows that "WiFi Digital Meter (unsupported)", so it doesn't list any entities
- And I also renewed my subscription to Tuya IoT
- And many-many restarts and Local Tuya reinstallations

**Comment by shazn786** [comment]
i have switched to standard toya integration by Home Assistant and problem
resolved. Now the WiFi breaker with energy meter works fine and all the
entities are displayed ie current, voltage and energy for phase A, B, C
(though B and C have zero values as this is a single phase device, so i
disabled the entities for B and C)

**Comment by gh69** [comment]
Just so there's no confusion...the problem technically isn't _resolved_, as the "no datapoints found" issue is with **LocalTuya**.  The standard Tuya integration does work for many of us but it requires that cloud connection, which is what LocalTuya helps us get away from.

**Comment by shazn786** [comment]
yes you are right

**Comment by deimosfr** [comment]
I'm geting the same issues :(

**Comment by xZetsubou** [comment]
There are 2 main reason that can cause this error, as far that I know
1. That you chose the wrong protocol version.
2. The device is busy due to another client connecting to the device, e.g [ Tuya app, smart life app and other local integration ]
  you need to make sure to the device you are trying to set it up isn't used by any other client.

Some said restart router or HA can fix this issue is probably because by restarting those u probably disconnected  the client that connected to the device.

I can't be sure enough I haven't used this repo since months [I use my fork ](https://github.com/xZetsubou/hass-localtuya) tho I haven't modify `pytuya` yet.

**Comment by lcsneil** [comment]
> There are 2 main reason that can cause this error, as far that I know
> 
> 1. That you chose the wrong protocol version.
> 2. The device is busy due to another client connected and using the device, e.g [ Tuya app, smart life app and other local integration ]  `Don't remove them just close those apps` you need to make sure to the device you are trying to set it up isn't used by any other client also isn't used locally in another HA instance.
> 
> Some said restart router or HA can fix this issue is probably because by restarting those u probably disconnected the client that connected to the device.
> 
> I can't be sure enough I haven't used this repo since months [I use my fork ](https://github.com/xZetsubou/hass-localtuya) tho I haven't modify `pytuya` yet.

Tried removing localtuya, rebooting, install this fork reboot and and got exactly the same issue (with all 4 protocol versions). No datapoints found.

lcsneil

**Comment by xZetsubou** [comment]
> > There are 2 main reason that can cause this error, as far that I know
> > 
> > 1. That you chose the wrong protocol version.
> > 2. The device is busy due to another client connected and using the device, e.g [ Tuya app, smart life app and other local integration ]  `Don't remove them just close those apps` you need to make sure to the device you are trying to set it up isn't used by any other client also isn't used locally in another HA instance.
> > 
> > Some said restart router or HA can fix this issue is probably because by restarting those u probably disconnected the client that connected to the device.
> > I can't be sure enough I haven't used this repo since months [I use my fork ](https://github.com/xZetsubou/hass-localtuya) tho I haven't modify `pytuya` yet.
> 
> Tried removing localtuya, rebooting, installed xZetsubou' fork reboot and and got exactly the same issue (with all 4 protocol versions). No datapoints found.
> 
> lcsneil

Are you also sure that your device protocol isn't `3.5` ? 
You can use the other [integration](https://github.com/make-all/tuya-local) for `3.5` devices it's relay on [tinytuya](https://github.com/jasonacox/tinytuya) so it's slower but  supports `3.5`. 
I don't have any `3.5` devices to make tests on.

**Comment by lcsneil** [comment]
Ok - I have solved it for me. End user error. Given that others sound like they have the same symptoms I will embarrass myself by admitting what I did wrong. Basically RTFM!  

I was using the 'secret' key on the Tuya cloud account, as the local key. This is NOT the one you need for the local key. That you find as per the instructions. Using that cloud key as the local one gives you device found but no data points as per my original problem. 

As the installations instructions say... 
**Under "Devices Management", select the "Query Device Details in Bulk" function, and enter your Device IDs, separated by commas. In the results you should see your local_key.**

Off to hang my head in shame ........

lcsneil
lcsneil

**Comment by martinvana** [comment]
Adding to the mix - in case someone else may benefit:
I was scratching my head for a while for the exactly same reason. In my case the integration pre-filled the **Local key** so I didn't doubt it.
However, _rebooting_ the entire HomeAssistant yielded a different **Local key** and that fixed the issue. :-)

**Comment by cooolinho** [comment]
Your Local Key can be found here:
Tuy IoT Platform > API Explorer > Device Management > Query Device Details in Bulk > PUT Your DEVICE ID > Press "Submit Request"

![image](https://github.com/rospogrigio/localtuya/assets/5195515/e52b0761-3bfd-4993-afc2-686d2dc31b55)

**Comment by rgnyldz** [comment]
I get this error message with 5.2.1 and Local key, ip address of the device, user id etc are all set up correctly. 

Tho in my old home assistant instance where I use 3.2.4.2 with the protocol version 3.5 everything is working fine. in 5.2.1 there is no 3.5 protocol version. 

Below is my log;

```
This error originated from a custom integration.

Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:863
integration: LocalTuya (documentation, issues)
First occurred: 2:30:06 AM (1 occurrences)
Last logged: 2:30:06 AM

[bfe...y6p] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
```

**Comment by JSferry** [comment]
Stò provando a configurare la centralina **_RainPoint TTV103FRF_wifi_** con localTuya ma ricevo anche io lo stesso errore:
> **_La connessione al dispositivo è riuscita ma non sono stati trovati i datapoint, riprova. Crea un nuovo Issue e includi i log di debug se il problema persiste._**

Quello che ho potuto notare è che il configuratore di localTuya trova solo l'HUB connesso alla rete.
Non posso aggiungere ne la valvola ne la presa perchè la loro **local_key** è vuota, l'unico ad averla è l' HUB.

Effettuando un controllo attraverso il pannello sviluppatore di Tuya IoT Platform, rispettivamente in **Cloud > Development > My Project > Devices** è possibile notare che l'HUB non ha nessun **DP instruction**, mentre la **Valvola di controllo dell'acuqa (Water valve controller**) e la **Presa (Socket**)" si, ma senza _local_key_ non riesco ad aggiungerle.

Qualcuno si trova nella mia stessa situazione ed è riuscito ad ovviare al problema in qualche modo??

**Comment by LangerFeen** [comment]
I had to select a lower protocol version 3.3 in the device config because the device wouldn't accept/respond to protocol version 3.4

**Comment by dombrowa** [comment]
Another cause can be that the devices key in the cloud changed (this happened to one of my devices)
"Error when adding in Local Tuya
Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.
"
apparently indicates that the device/endpoint is found (local tuya can find this IP and likely connect to ist servce) but not necessarily query the device because it needs a valid key to authenticate any request.
The error message is not clear IMHO.

**Comment by hmlendea** [comment]
I managed to solve this problem, at least on my end!

TL;DR: The local key I used was wrong.

The switch I tried to add was actually re-paired multiple times in the past. I am retrieving the local key by copying the app's XML from a rooted Android phone and copying it from there.
However this only works with a few specific versions of the app. I knew that, but what I didn't know is that the app updated itself without me noticing. And that file remained in the filesystem, it wasn't removed, so I didn't notice anything wrong since I was able to copy and parse it... it was just after I renamed the switch and copied the file for the thousandth time that I realised its not accurate, since it also displayed the old name. Reinstalling the app with the correct version and doing the proccess again one last time resulted in me getting the correct local key and being able to add it to localtuya successfully.

**Comment by korfox13** [comment]
Another hint: If you use german umlauts in the name of the device (e.g. Küchenlampe) the same error appears... replacing ü by ue solved it for me.

**Comment by diagpope** [comment]
I believe that the same message "Connection to device succeeded but no datapoints found" appears if the local integration does not support protocol v3.5 of newer Tuya devices. I have two Dual clamp energy meters working on v3.4 but the v3.5 one shows up in HomeAssistant with just one sensor,  but normal in the Tuya App and the Tuya Cloud API shows all sensors.

**Comment by WBrown15** [comment]
have the same problem, in smartlifeapp it works fine, tuya project as well detects all platforms, but in local tuya i get the error  no datapoints found.
  i used a python tinytuya (git project) script to scan for the local key . for 2 out of 4 devices it worked fine. 2 others i get the above mentioned error. so i assume my local keys are correct but i want double check that. i wondering as well on the ip /host to use. tuya cloud project shows ip addresses for devices which don't match either my external nor internal ip addresses. 

which host  do you use ?
is the name dependend on the name shown in cloud or can it be choosen by your own in home assistent?
where did you get the local key from? 
which tool did you use to get it?

any help is appreciated

**Comment by diagpope** [comment]
Host: Each of your Tuya device will have its own IP on your local network . You may have set them to static *dhcp* on your Wifi router. 
Name in HASS is free but the id string is determined by Tuya cloud afaik.
Since you have a device that says no datapoints, the endpoint/IP must exist (I assume) but is it a Tuya device?
if yes, check which FW it uses as it failed for me when 3.5 was removed from the integration (see Settings -> Devices -> Device info)
The local key I retrieved via  tuya-cli wizard  (I do  not remember how I installed it) or you can go into the Tuya Cloud
```
login
https://platform.tuya.com/cloud/....
Open Project
Smart Home
Cloud
 Api Explorer
  Device Control
    Query Properties  <some id string of the device>
```

**Comment by blacksocket** [comment]
The same issue here. All ip's and details are correct. See attached logs:



2025-06-25 20:03:00.267 DEBUG (MainThread) [custom_components.localtuya.discovery] Discovered device: {'ip': '192.168.1.205', 'gwId': 'bff7c2c202bdcd338do565', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'gp3tq6ifgxwky4dq', 'version': '3.3'}
2025-06-25 20:03:00.267 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:05.269 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:10.283 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:15.281 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:20.282 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:25.300 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:30.298 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:34.334 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'localtuya' sets option flow config_entry explicitly, which is deprecated at custom_components/localtuya/config_flow.py, line 396: self.config_entry = config_entry. This will stop working in Home Assistant 2025.12, please create a bug report at https://github.com/rospogrigio/localtuya/issues
2025-06-25 20:03:35.299 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:40.311 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:45.312 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:50.315 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:03:55.325 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:00.323 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:05.327 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:10.333 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:11.868 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Sending command 10 (device type: type_0a)
2025-06-25 20:04:11.868 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Sending payload: b'{"gwId":"bfa34bb861b1429ae4slpc","devId":"bfa34bb861b1429ae4slpc","uid":"bfa34bb861b1429ae4slpc","t":"1750874651"}'
2025-06-25 20:04:11.869 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Command 10 waiting for seq. number 1
2025-06-25 20:04:15.336 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:16.871 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Command 10 timed out waiting for sequence number 1
2025-06-25 20:04:16.871 ERROR (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:04:16.873 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:04:16.873 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:04:16.873 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:04:16.873 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Closing connection
2025-06-25 20:04:16.874 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Connection lost: None
2025-06-25 20:04:20.338 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:25.346 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:30.348 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:35.350 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:40.358 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:45.361 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:50.362 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:54.772 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Sending command 10 (device type: type_0a)
2025-06-25 20:04:54.772 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Sending payload: b'{"gwId":"bfa34bb861b1429ae4slpc","devId":"bfa34bb861b1429ae4slpc","uid":"bfa34bb861b1429ae4slpc","t":"1750874694"}'
2025-06-25 20:04:54.772 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Command 10 waiting for seq. number 1
2025-06-25 20:04:55.374 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:04:59.773 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Command 10 timed out waiting for sequence number 1
2025-06-25 20:04:59.773 ERROR (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:04:59.775 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:04:59.775 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:04:59.775 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:04:59.775 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Closing connection
2025-06-25 20:04:59.775 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Connection lost: None
2025-06-25 20:05:00.378 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:05.380 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:10.389 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:15.391 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:18.513 ERROR (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:05:18.514 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:05:18.514 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:05:18.514 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:05:20.392 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:25.401 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:30.405 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:34.385 ERROR (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:05:34.386 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:05:34.386 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:05:34.386 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:05:35.405 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:40.414 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:45.416 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:50.418 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:55.427 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:05:59.130 ERROR (MainThread) [custom_components.localtuya.pytuya] [bfa...lpc] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:05:59.131 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:05:59.131 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:05:59.131 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:06:00.429 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:05.431 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:10.443 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:15.445 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:20.447 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:25.456 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:30.458 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:35.460 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:40.470 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:45.472 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:50.474 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:06:55.483 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:00.485 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:05.487 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:10.499 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:15.502 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:20.503 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:25.512 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:30.514 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:30.817 ERROR (MainThread) [custom_components.localtuya.pytuya] [bf0...77g] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:07:30.818 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:07:30.818 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:07:30.818 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:07:35.516 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:40.527 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:45.528 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:50.530 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:55.538 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:07:58.714 ERROR (MainThread) [custom_components.dahua] 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 347, in _wait
    await waiter
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/dahua/client.py", line 744, in stream_events
    async for data, _ in response.content.iter_chunks():
        on_receive(data, channel)
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 71, in __anext__
    rv = await self._stream.readchunk()
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 476, in readchunk
    await self._wait("readchunk")
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 346, in _wait
    with self._timer:
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/aiohttp/helpers.py", line 685, in __exit__
    raise asyncio.TimeoutError from exc_val
TimeoutError
2025-06-25 20:08:00.541 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:05.542 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:10.551 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:13.480 ERROR (MainThread) [custom_components.localtuya.pytuya] [bf0...77g] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:08:13.480 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:08:13.480 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:08:13.481 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:08:15.553 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:20.555 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:25.568 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:30.570 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:35.572 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:37.478 ERROR (MainThread) [custom_components.localtuya.pytuya] [bf0...77g] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:08:37.479 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:08:37.479 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:08:37.479 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:08:40.580 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:45.583 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:50.584 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:08:52.837 ERROR (MainThread) [custom_components.localtuya.pytuya] [bf0...77g] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 507, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/locks.py", line 407, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.13/asyncio/tasks.py", line 506, in wait_for
    async with timeouts.timeout(timeout):
               ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/timeouts.py", line 116, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2025-06-25 20:08:52.838 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2025-06-25 20:08:52.838 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2025-06-25 20:08:52.838 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2025-06-25 20:08:55.593 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:09:00.652 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:09:05.597 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565
2025-06-25 20:09:10.606 DEBUG (MainThread) [custom_components.localtuya.common] Missing device configuration for device_id bff7c2c202bdcd338do565

**Comment by diagpope** [comment]
@blacksocket 

Your logs shows 75 repeated errors of:  'Missing device configuration for device_id bff7c2c202bdcd338do565' 
Stating the obvious: Have you addressed the issue that HASS complains about the device `bff7c2c202bdcd338do565` not being configured? Have you gone thru the setup and given all the sensor specific DPS entries (name, unit, scaling, etc.)

You should have something like this showing as your sensor in your HASS instance: /homeassistant/.storage/core.config_entries

```
             "friendly_name": "my sensor",
              "host": "my ip",
              "local_key": "secret key",
              "manual_dps_strings": "101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,122,123,124",
              "node_id": null,
              "product_key": "<product key>",
              "protocol_version": "3.5",
              "scan_interval": 20
            },
            "my device id": {
              "device_id": "my device id",
              "dps_strings": [
                "101 ( value: 7046 )",
                "102 ( value: REVERSE )",
                "104 ( value: REVERSE )",
                "105 ( value: 2802 )",
                "106 ( value: 10685 )",
                "107 ( value: 174994 )",
                "108 ( value: 146573 )",
                "109 ( value: 110902 )",
                "110 ( value: 97 )",
                "111 ( value: 5884 )",
                "112 ( value: 1236 )",
                "113 ( value: 5824 )",
                "114 ( value: 3685 )",
                "115 ( value: -9848 )",
                "116 ( value: 1000 )",
                "117 ( value: 1000 )",
                "118 ( value: 1000 )",
                "119 ( value: 1000 )",
                "121 ( value: 65 )",
                "122 ( value: 1000 )",
                "123 ( value: 1071 )",
                "124 ( value: 1000 )",
                "125 ( value: 1000 )",
                "127 ( value: 1000 )",
                "128 ( value: 1000 )",
                "129 ( value: 10 )",
                "130 ( value: 157258 )",
                "131 ( value: 285896 )"
              ],
              "enable_debug": false,
              "entities": [
 ```

Although HASS says:
```
DEBUG (MainThread) [custom_components.localtuya.discovery] Discovered device: {'ip': '192.168.1.205', 'gwId': 'bff7c2c202bdcd338do565', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'gp3tq6ifgxwky4dq', 'version': '3.3'}
```
you may have not given the DPS or any other data required to query the device.

Does the device show in the Tuya cloud with 192.168.1.205, bff7c2c202bdcd338do565, gp3tq6ifgxwky4dq running 3.3?

**Comment by blacksocket** [comment]
I added all of them including DPSID and still the same issue. Strange that for other swiches it works only not for the new I bought last time. Additionaly for the new one it works also for normal Tuya integration but after last HA upgrade it stopped working. That is why i wanted to move to the local Tuya integration.

https://www.temu.com/pl/goods.html?_bg_fs=1&goods_id=601099598343043&sku_id=17592524468049&_x_sessn_id=v7f74mq9ow&refer_page_name=bgt_order_detail&refer_page_id=10045_1751022378544_fet50338dv&refer_page_sn=10045


```
Rejestrator: custom_components.localtuya.common
Źródło: custom_components/localtuya/pytuya/__init__.py:255
integracja: LocalTuya integration ([dokumentacja](https://github.com/rospogrigio/localtuya/), [Problemy](https://github.com/rospogrigio/localtuya/issues))
Pierwsze zdarzenie: 12:49:36 (22 wystąpienia)
Ostatnio zalogowany: 12:55:55

[bfa...lpc] Disconnected - waiting for discovery broadcast
[032...00d] Disconnected - waiting for discovery broadcast
[611...65f] Disconnected - waiting for discovery broadcast
[bff...565] Disconnected - waiting for discovery broadcast
```


pi@raspberrypi:~ $ tinytuya devices

TinyTuya (Tuya device scanner) [1.17.2]

Loaded devices.json - 1 devices:


Device Listing

[
    {
        "name": "Salon",
        "id": "bfa34bb861b1429ae4slpc",
        "key": "xxxxxxxxx",
        "ip": "192.168.1.215",
        "version": "3.3"
    }
]

Poll local devices? (Y/n): y

Scanning local network for Tuya devices...
    2 local devices discovered

Polling local devices...
    [Salon                    ] 192.168.1.215      - [Off] - DPS: {'1': False, '9': 0, '38': 'memory', '39': 'relay', '40': False, '43': ''}

>> Saving device snapshot data to snapshot.json


Done.

**Comment by vgdh** [comment]
same error
this in logs:
```
2025-10-20 22:26:59.436 ERROR (MainThread) [homeassistant] Error doing job: Exception in callback _SelectorDatagramTransport._read_ready() (None)
Traceback (most recent call last):
  File "/config/custom_components/localtuya/discovery.py", line 65, in datagram_received
    data = decrypt_udp(data)
  File "/config/custom_components/localtuya/discovery.py", line 30, in decrypt_udp
    return _unpad(decryptor.update(message) + decryptor.finalize()).decode()
                                              ~~~~~~~~~~~~~~~~~~^^
ValueError: The length of the provided data is not a multiple of the block length.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/asyncio/selector_events.py", line 1244, in _read_ready
    self._protocol.datagram_received(data, addr)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/config/custom_components/localtuya/discovery.py", line 67, in datagram_received
    data = data.decode()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x90 in position 0: invalid start byte
```

**Comment by nktnet1** [comment]
Same issue for me for a light bulb. For some reason, the issue goes away when switching to this fork:

- https://github.com/xZetsubou/hass-localtuya

(and using protocol version 3.4/3.5)

**Comment by Konsts** [comment]
Same Issu with MakeSkyBlue mppt controller, device not connected

**Comment by jtokach1** [comment]
Same problem with 
Minoston Countdown Timer Switch
